core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/AbstractUserStoreManager.java

```diff
@@ -7947,7 +7947,16 @@ public void addSystemRole(String roleName, String[] userList, Permission[] permi
             return (String[]) object;
         }
 
-        return getUserRoles(username, filter);
+        if (isUniqueUserIdEnabledInUserStore(getUserStore(username))) {
+            String userID = getUserIDFromUserName(username);
+            if (userID == null) {
+                // According to implementation, getRoleListOfUser method would return everyone role name for all users.
+                return new String[]{realmConfig.getEveryOneRoleName()};
+            }
+            return getUserRolesWithID(userID, "*").toArray(new String[0]);
+        } else {
+            return getUserRoles(username, "*");
```

Comment:
Shouldn't we use `filter` instead of `*`

Reply:
Resolved in https://github.com/wso2/carbon-kernel/pull/2512/commits/d157f3092249071c71426f62170254838b9f36a3